documentation/grammars.md

```diff
@@ -0,0 +1,62 @@
+# Grammar Syntax Documentation
+
+Grammars in Pliant are defined by an bnf dialect similar to abnf and ebnf. Some additional control statements have been added for ignoring whitespace and specifying start productions.
+
+# Grammar of Grammars
+
+The supported language for grammars in Pliant is defined by the following BNF grammar. 
+
+```
+Definition          =   Block
+                        | Block Definition
+
+Block               =   Rule
+                        | Setting
+                        | LexerRule
+
+Rule                =   QualifiedIdentifier '=' Expression ';'
+
+Setting             =   SettingIdentifier '=' QualifiedIdentifier ';'
+
+LexerRule           =   QualifiedIdentifier '~' LexerRuleExpression ';'
+
+Expression          =   Term
```

Comment:
What are the semantics for '~'? I looked at the RFC documents and didn't see it defined anywhere. 

In the pliant definition language the tilde defines a "lexer rule".

I remember the '-' defines a except rule ex: A -B  == match A and Not B